driver/src/main/java/org/neo4j/Driver.java

```diff
@@ -0,0 +1,192 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.neo4j;
+
+import java.net.URI;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.neo4j.driver.Config;
+import org.neo4j.driver.Session;
+import org.neo4j.driver.Value;
+import org.neo4j.driver.Values;
+import org.neo4j.driver.exceptions.ClientException;
+import org.neo4j.driver.internal.StandardSession;
+import org.neo4j.driver.internal.pool.StandardConnectionPool;
+import org.neo4j.driver.internal.spi.ConnectionPool;
+import org.neo4j.driver.internal.spi.Logging;
+
+/**
+ * A driver to a Neo4j database is a connection manager to the database.
+ * It provides methods to establish {@link Session sessions} with a Neo4j instance to run statements.
+ * <P>
+ * A driver to a Neo4j instance could be created by providing a url to the database and a {@link Config} for any user
+ * specified configuration. When a user finishes work with the database, he/she should release the resources used by
+ * this driver by invoking {@code driver.close()}
+ * <p>
+ * Example:
+ * <pre>
+ * {@code
+ *
+ * // Create a driver with default configuration
+ * Driver driver = GraphDatabase.driver( "neo4j://localhost:7687" );
+ *
+ * // Establish a session with a Neo4j instance
+ * Session session = driver.session();
+ * ** Do some work with the database... **
+ *
+ * // Release all the resources
+ * driver.close();
+ * }
+ * </pre>
+ * <p>
+ * After a session is established, we could run statements
+ * <p>
+ * For example:
+ * <pre>
+ * {@code
+ *
+ * // Run a single statement
+ * session.run( "CREATE (n {name:'Bob'})" );
+ *
+ * // Run multiple statements in a transaction
+ * try( Transaction tx = session.newTransaction() )
+ * {
+ *     tx.run( "CREATE (n {name:'Alice'})" );
+ *     tx.run( "CREATE (n {name:'Tina'})" );
+ *     tx.success();
+ * }
+ *
+ * // Retrieve results from a query
+ * Result result = session.run( "MATCH (n) RETURN n.name" );
+ * while( result.hasNext())
+ * {
+ *     Value record = result.next();
+ *     System.out.println( record.get("n.name") );
+ * }
+ * }
+ * </pre>
+ * <p>
+ * A driver manages a connection pool to a Neo4j instance.
+ * Therefore more than one session could be established over one driver until the driver's connection pool is full.
+ * (The default connection pool size could be found in {@code Config}.)
+ * A connection could be returned to the connection pool by invoking {@link Session#close() close} when the work over
+ * a session is done.
+ */
+public class Driver implements AutoCloseable
+{
+    /**
+     * A static class to create a driver
+     */
+    public static class GraphDatabase
```

Comment:
I would want to move this out, so it's not an inner class; that way we get:

    Driver driver = GraphDatabase.driver(..);

instead of:

    Driver driver = Driver.GraphDatabase.driver(..);